src/Core/Services/MetadataProviders/MetadataProviderFactory.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO.Abstractions;
+using System.Net;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Azure.DataApiBuilder.Core.Configurations;
+using Azure.DataApiBuilder.Core.Resolvers.Factories;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.DataApiBuilder.Core.Services.MetadataProviders
+{
+    /// <inheritdoc />
+    public class MetadataProviderFactory : IMetadataProviderFactory
+    {
+        private readonly IDictionary<string, ISqlMetadataProvider> _metadataProviders;
+
+        public MetadataProviderFactory(RuntimeConfigProvider runtimeConfigProvider, IAbstractQueryManagerFactory engineFactory, ILogger<ISqlMetadataProvider> logger, IFileSystem fileSystem)
```

Comment:
nit: shouldn't name of the argument `engineFactory` actually be `queryManagerFactory`?

Reply:
Fixed.